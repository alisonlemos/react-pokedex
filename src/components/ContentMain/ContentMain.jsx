import CategoriesFilter from './components/CategoriesFilter';
import FavoritesFilter from './components/FavoritesFilter';
import PokemonList from './components/PokemonList';
import './ContentMain.scss';

function ContentMain() {
	return (
		<section className="contentMain__section">
			<aside>
				<CategoriesFilter />
				<FavoritesFilter />
			</aside>
			<PokemonList />
		</section>
	);
}

export default ContentMain;
