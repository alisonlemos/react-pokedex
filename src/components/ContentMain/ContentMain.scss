@import '../../assets/colors.scss';

.contentMain__section {
	display: flex;
	aside {
		display: flex;
		flex-direction: column;
		width: 15%;
	}
}
