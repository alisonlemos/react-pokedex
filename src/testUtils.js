import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { orderValues } from './services';
import { setupStore } from './store';

export const mockedPokemonList = {
	data: {
		results: [
			{
				national_number: '99999',
				sprites: {
					normal: 'normal_sprite_url',
				},
				name: 'Pokemon Name',
				type: ['Type_1', 'Type_2'],
			},
		],
	},
};

export const initialState = {
	pokemonList: [],
	favoriteList: [],
	categoriesList: [],
	searchKey: '',
	order: orderValues.NATIONAL_NUMBER_ASC,
	favoriteFilter: false,
	categoriesFilter: [],
	loading: false,
};

export function renderWithProviders(
	ui,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}

	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
