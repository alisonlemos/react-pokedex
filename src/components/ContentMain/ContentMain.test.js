import { renderWithProviders } from '@/testUtils';
import ContentMain from './ContentMain';

describe('Componente ContentMain', () => {
	it('deve renderizar o componente de acordo com o snapshot', () => {
		expect(renderWithProviders(<ContentMain />)).toMatchSnapshot();
	});
});
