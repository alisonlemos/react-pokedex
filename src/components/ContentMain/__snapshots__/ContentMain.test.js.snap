// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente ContentMain deve renderizar o componente de acordo com o snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="contentMain__section"
      >
        <aside>
          <div
            class="categoriesFilter__container"
          >
            <h3>
              Filtrar por tipo
            </h3>
          </div>
          <div
            class="favoriteFilter__container"
          >
            <h3>
              Filtrar Favoritos
            </h3>
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium favoriteFilter__switch css-julti5-MuiSwitch-root"
            >
              <span
                class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  aria-label="fitrar favoritos"
                  class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                  type="checkbox"
                />
                <span
                  class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
          </div>
        </aside>
        <div
          class="pokemonList__container"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="contentMain__section"
    >
      <aside>
        <div
          class="categoriesFilter__container"
        >
          <h3>
            Filtrar por tipo
          </h3>
        </div>
        <div
          class="favoriteFilter__container"
        >
          <h3>
            Filtrar Favoritos
          </h3>
          <span
            class="MuiSwitch-root MuiSwitch-sizeMedium favoriteFilter__switch css-julti5-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                aria-label="fitrar favoritos"
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
        </div>
      </aside>
      <div
        class="pokemonList__container"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
